src/sign.c

```diff
@@ -457,8 +457,10 @@ buf_delsign(
     // When deleting the last sign the cursor position may change, because the
     // sign columns no longer shows.
     if (buf->b_signlist == NULL)
+    {
+	redraw_buf_later(buf, NOT_VALID);
```

Comment:
Perhaps we can do better here.  I think we'd only need to cal redraw_buf_later(…) when 'signcolumn' option is equal to "auto".